packages/application-config/src/process-config.ts

```diff
@@ -28,21 +40,182 @@ type ProcessConfigOptions = Partial<LoadingConfigOptions> & {
+};
+
 const trimTrailingSlash = (value: string) => value.replace(/\/$/, '');
 
-const omitDevConfigIfEmpty = (
-  devConfig: ApplicationRuntimeConfig['env']['__DEVELOPMENT__']
-) => {
+const omitDevConfigIfEmpty = <Config>(devConfig: Config) => {
   if (
-    // @ts-expect-error: the `accountLinks` is not explicitly typed as it's only used by the account app.
-    devConfig?.accountLinks ||
-    devConfig?.menuLinks ||
-    devConfig?.oidc
-  )
+    devConfig &&
+    (Object.hasOwn(devConfig, 'accountLinks') ||
+      Object.hasOwn(devConfig, 'menuLinks') ||
+      Object.hasOwn(devConfig, 'customViewHostUrl') ||
+      Object.hasOwn(devConfig, 'oidc'))
+  ) {
     return devConfig;
+  }
   return undefined;
 };
 
+const isCustomViewData = (
+  data: CustomApplicationData | CustomViewData
+): data is CustomViewData =>
+  (data as CustomApplicationData).entryPointUriPath === undefined;
+
+const getRuntimeEnvironmentConfigForDevelopment = ({
+  isProd,
+  configurationData,
+  mcApiUrl,
+  appConfig,
+  entryPointUriPath,
+}: {
+  isProd: boolean;
+  configurationData: CustomApplicationData | CustomViewData;
+  mcApiUrl: URL;
+  appConfig:
+    | JSONSchemaForCustomApplicationConfigurationFiles
+    | JSONSchemaForCustomViewConfigurationFiles;
+  entryPointUriPath: string;
+}): ApplicationRuntimeEnvironmentForDevelopment | undefined => {
+  if (isProd) {
+    return undefined;
+  }
+
+  const oidcConfig = omitEmpty<ApplicationOidcForDevelopmentConfig>({
+    authorizeUrl: [
+      // In case the MC API url points to localhost, we need to point
+      // to a local running dev login page to handle the workflow properly.
+      mcApiUrl.hostname === 'localhost'
+        ? mcApiUrl.origin.replace(mcApiUrl.port, String(developmentPort))
+        : mcApiUrl.origin.replace('mc-api', 'mc'),
+      '/login/authorize',
+    ].join(''),
+    initialProjectKey:
+      // For the `account` application, we should unset the projectKey.
+      entryPointUriPath === 'account'
+        ? undefined
+        : appConfig.env.development.initialProjectKey,
+    ...(appConfig.env.development?.teamId && {
+      teamId: appConfig.env.development.teamId,
+      // When using a teamId we need to send the actual application of view identifier.
+      ...(isCustomViewData(configurationData)
+        ? { customViewId: configurationData.id }
+        : { applicationId: configurationData.id }),
+    }),
+    oAuthScopes: appConfig.oAuthScopes,
+    additionalOAuthScopes: appConfig?.additionalOAuthScopes,
+  });
+
+  if (isCustomViewData(configurationData)) {
+    const hostUriPath = (appConfig as JSONSchemaForCustomViewConfigurationFiles)
+      .env.development.hostUriPath;
+    const defaultHostUriPath = oidcConfig.initialProjectKey
+      ? `/${oidcConfig.initialProjectKey}/${entryPointUriPath}`
+      : `/${entryPointUriPath}`;
+    const hostUrl = new URL(
+      hostUriPath || defaultHostUriPath,
+      'http://localhost:3001'
+    );
```

Comment:
For development we construct the host URL programmatically, unless the URI path is explicitly provided in the config file